testng-core/src/main/java/org/testng/TestNG.java

```diff
@@ -343,12 +343,8 @@ private OverrideProcessor getProcessor() {
   private Collection<XmlSuite> processCommandLineArgs(Collection<XmlSuite> allSuites) {
     Collection<XmlSuite> result = new ArrayList<>();
     for (XmlSuite s : allSuites) {
-      if (this.m_parallelMode != null) {
-        s.setParallel(this.m_parallelMode);
-      }
-      if (this.m_threadCount > 0) {
-        s.setThreadCount(this.m_threadCount);
-      }
+      processParallelModeCommandLineArgs(s);
+      s.getChildSuites().forEach(this::processParallelModeCommandLineArgs);
```

Comment:
Child suites of child suites won't be configured. But I can't remember if TestNG supports child-child-suites.

Reply:
make `processParallelModeCommandLineArgs(s);`  recursive